edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/config/TwinConfigSource.cs

```diff
@@ -186,6 +232,42 @@ Task UpdateReportedProperties(long desiredVersion, LastDesiredStatus desiredStat
             }
         }
 
+        private static bool CheckIfTwinPropertiesAreSigned(TwinCollection twinDesiredProperties) => JObject.Parse(twinDesiredProperties.ToString())["integrity"] != null;
+
+        public static bool ExtractHubTwinAndVerify(TwinCollection twinDesiredProperties)
```

Comment:
This one is pretty much the same as the one for edgeagent. Please see if we can have a utility function and reuse.

Reply:
its not the same. Edgehub and EA has different schema for desired properties. so we need to extract individual components and verify which is different. for ex. EA has modules, runtime, systemmodules and SchemaVersion and EH has schemaversion, routes and storeAndForwardConfiguration.